SimPEG/ObjectiveFunction.py

```diff
@@ -0,0 +1,258 @@
+
+    def __div__(self, denominator):
+        return self.__mul__(1./denominator)
+
+
+class ComboObjectiveFunction(BaseObjectiveFunction):
+
+    _multiplier_types = (float, None, Utils.Zero) + integer_types # Directive
+
+    def __init__(self, objfcts=[], multipliers=None, **kwargs):
+
+        self.objfcts = []
+        nP = '*'
+
+        for fct in objfcts:
+            assert isinstance(fct, BaseObjectiveFunction), (
+                "Unrecognized objective function type {} in objfcts. All "
+                "entries in objfcts must inherit from "
+                "ObjectiveFunction".format(fct.__class__.__name__)
+            )
+
+            # ensure all objective functions have the same nP
+            if fct.nP != '*':
+                if nP != '*':
+                    assert nP == fct.nP, (
+                        "Objective Functions must all have the same nP, not "
+                        "{}".format([f.nP for f in objfcts])
+                    )
+                else:
+                    nP = fct.nP
+            self.objfcts.append(fct)
+
+        if multipliers is None:
+            multipliers = len(self.objfcts)*[1]
+        else:
+            for mult in multipliers:
+                assert(type(mult) in self._multiplier_types), (
+                    "Objective Functions can only be multiplied by a float, or"
+                    " a properties.Float, not a {}, {}".format(
+                        type(mult), mult
+                    )
+                )
+            assert len(multipliers) == len(self.objfcts), (
+                "Length of multipliers ({}) must be the same as the length of "
+                "objfcts ({})".format(len(multipliers), len(self.objfcts))
+            )
+        self._multipliers = multipliers
+
+        super(ComboObjectiveFunction, self).__init__(**kwargs)
+        self._nP = nP
+
+    @property
+    def multipliers(self):
+        return self._multipliers
+
+    def _eval(self, x, **kwargs):
+        f = Utils.Zero()
+        for multpliter, objfct in zip(self.multipliers, self.objfcts):
+            f += multpliter * objfct(x, **kwargs)
+        return f
+
+    def deriv(self, x, **kwargs):
+        g = Utils.Zero()
+        for multpliter, objfct in zip(self.multipliers, self.objfcts):
+            g += multpliter * objfct.deriv(x, **kwargs)
+        return g
+
+    def deriv2(self, x, **kwargs):
+        H = Utils.Zero()
+        for multpliter, objfct in zip(self.multipliers, self.objfcts):
+            objfct_H = objfct.deriv2(x, **kwargs)
+            if isinstance(objfct_H, Utils.Zero):
+                pass
+            elif isinstance(objfct_H, Utils.Identity) and self.nP != '*':
+                H += multpliter * sp.eye(self.nP)  # if we need a shape
+            else:
+                H += multpliter * objfct_H
+        return H
+
+    # This assumes all objective functions have a W.
+    # The base class currently does not.
+    @property
+    def W(self):
```

Comment:
no. I thought it might be handy to be able to interact with, but it is not actually used anywhere in the code